smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/CodegenVisitor.java

```diff
@@ -429,7 +430,7 @@ private void generateClient(ServiceShape shape) {
         for (OperationShape operation : containedOperations) {
             if (operation.hasTrait(PaginatedTrait.ID)) {
                 hasPaginatedOperation = true;
-                String outputFilename = PaginationGenerator.getOutputFilelocation(operation);
+                String outputFilename = CodegenUtils.SOURCE_FOLDER + "/" + PaginationGenerator.getOutputFilelocation(operation);
```

Comment:
All these get output file location operations / constants should embed this prefix rather than you adding it on in post.

Reply:
If I update output file location, the import in `index.ts` imports from `src` instead directly.
* Original commit which added prefix in output file location [`34658f6` (#432)](https://github.com/awslabs/smithy-typescript/pull/432/commits/34658f65b7f9b9df6e6b308d222dae2dfd1dcc1f)
* It was moved outside to fix exports in index.ts [`4d9d063` (#432)](https://github.com/awslabs/smithy-typescript/pull/432/commits/4d9d0639bbb69218f044cfb24506a3c2b6d0812b)

Should I remove `src` while exporting from IndexGenerator instead?
Or use some abstraction which removes `src` automatically?

Reply:
You could have a `getOutputFilelocation` and `getOutputNamespace` or some such, the latter of which would not include src.

Reply:
Done in [`53b6977` (#432)](https://github.com/awslabs/smithy-typescript/pull/432/commits/53b69774a11fd93412825cb3460b036cd35f50c4)